### PR TITLE
[0-size Tensor support] fix std in some 0-size case

### PR DESCRIPTION
**Pull request by DanielSun11**
### PR Category

Execute Infrastructure

### PR Types

Bug fixes 

### Description

从PaddleAPITest报错：
```
2025-06-08 12:06:59.072995 GPU 3 63652 test begin: paddle.std(Tensor([],"float32"), )
W0608 12:06:59.222908 63652 dygraph_functions.cc:87666] got different data type, run type promotion automatically, this may cause data type been changed.
/root/miniconda3/envs/paddle/lib/python3.12/site-packages/paddle/tensor/stat.py:285: UserWarning: Degrees of freedom is <= 0.
  out = var(**locals())
[accuracy error] paddle.std(Tensor([],"float32"), ) 
 Not equal to tolerance rtol=0.01, atol=0.01
Scalars are not close!

Expected 0.0 but got nan.
Absolute difference: nan (up to 0.01 allowed)
Relative difference: inf (up to 0.01 allowed)
ACTUAL: (shape=torch.Size([]), dtype=torch.float32)
nan
DESIRED: (shape=torch.Size([]), dtype=torch.float32)
0.0

2025-06-08 12:06:59.102253 GPU 4 63646 test begin: paddle.std(Tensor([],"float32"), list[], )
W0608 12:06:59.242542 63646 dygraph_functions.cc:87666] got different data type, run type promotion automatically, this may cause data type been changed.
/root/miniconda3/envs/paddle/lib/python3.12/site-packages/paddle/tensor/stat.py:285: UserWarning: Degrees of freedom is <= 0.
  out = var(**locals())
[accuracy error] paddle.std(Tensor([],"float32"), list[], ) 
 Not equal to tolerance rtol=0.01, atol=0.01
Scalars are not close!

Expected 0.0 but got nan.
Absolute difference: nan (up to 0.01 allowed)
Relative difference: inf (up to 0.01 allowed)
ACTUAL: (shape=torch.Size([]), dtype=torch.float32)
nan
DESIRED: (shape=torch.Size([]), dtype=torch.float32)
0.0
```
上述错误中torch的返回结果为0.0，paddle的返回结果为nan。但是通过
```python
import torch
import numpy as np

shape1 = []  

tensor1 = torch.tensor([])

print("Torch Tensor 1 shape:", tensor1.shape)
torch_result = torch.std(tensor1)

print("Torch Result shape :", torch_result.shape)
print("Torch Result :", torch_result)

import paddle
tensor1 = paddle.to_tensor([], dtype='float32')
paddle_result = paddle.std(tensor1)

print("paddle Result shape :", paddle_result.shape)
print("paddle Result  :", paddle_result)
np.testing.assert_allclose(torch_result.detach().numpy(), paddle_result.numpy(), rtol=1e-05)
```

```
# outputs
Torch Result shape : torch.Size([])
Torch Result : tensor(nan)

paddle Result shape : []
paddle Result  : Tensor(shape=[], dtype=float32, place=Place(gpu:0), stop_gradient=True,
       nan)
```
测试输出都是正常的。
- 仅发现当输入为一个标量tensor(0-dim)并且unbiased（torch为correction）参数为False时，torch返回0，paddle返回nan。修复了上述问题。
- 修复后上述程序可以通过，PaddleAPITest仍然报错。
- 添加了var和std的单测case
pcard-67164

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM
